### ISSUE TITLE
depexts: No versioning

### ISSUE THREAD

**Issue by juergenhoetzel** [ocaml/opam#2168]
From the Manual

     `ext-packages` should be identifiers of packages recognised 

It would be nice to have  version constraints like in the `depends` field to prevent build failures because of outdated or incompatible versions

**Comment by AltGr** [comment]
Indeed; these would be difficult to map to the underlying package management system though (and most distributions only provide one version of each package)

**Comment by tbrk** [comment]
Another motivation for version constraints on *depexts* would be to handle packing changes across different releases. For example, inria-parkas/sundialsml requires *libsundials-serial-dev* on Debian Stretch (stable), but *libsundials-dev* on Debian Sid (unstable).

**Comment by kit-ty-kate** [comment]
@tbrk opam2 can handle this I think (see https://github.com/ocaml/opam/issues/2919)

I think this issue can be closed

**Comment by ejgallego** [comment]
Hi folks,

if my understanding of the depext situation is correct I'd like to propose this issue is not yet closed. IMO depext should support versioning on the package itself, and the version should not be the os-version but the upstream package version.

For example, in lablgtk we should be able to do:
```
depexts: [ "gtk3.0" >= "3.12" ]
```
etc... then, `depext` shoud be able to query the underlying os for the upstream version provided.
